float64 x_pos
float64 y_pos
float64 z_pos
float64 x_ang
float64 y_ang
float64 z_ang
